query/sources/impl/src/test/groovy/org/codice/ddf/admin/sources/csw/persist/SaveCswConfigurationTest.groovy

```diff
@@ -0,0 +1,309 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.sources.csw.persist
+
+import ddf.catalog.source.FederatedSource
+import org.codice.ddf.admin.api.action.Action
+import org.codice.ddf.admin.common.actions.BaseAction
+import org.codice.ddf.admin.common.message.DefaultMessages
+import org.codice.ddf.admin.configurator.ConfigReader
+import org.codice.ddf.admin.configurator.Configurator
+import org.codice.ddf.admin.configurator.ConfiguratorFactory
+import org.codice.ddf.admin.configurator.OperationReport
+import org.codice.ddf.admin.sources.commons.SourceMessages
+import org.codice.ddf.admin.sources.fields.CswProfile
+import org.codice.ddf.admin.sources.fields.CswSpatialOperator
+import org.codice.ddf.admin.sources.fields.type.CswSourceConfigurationField
+import spock.lang.Specification
+
+import static org.codice.ddf.admin.sources.SourceTestCommons.*
+
+class SaveCswConfigurationTest extends Specification {
+
+    static TEST_OUTPUT_SCHEMA = 'testOutputSchema'
+
+    static TEST_CSW_PROFILE = CswProfile.CSW_FEDERATION_PROFILE_SOURCE
+
+    static CSW_PROFILE = CswProfile.DEFAULT_FIELD_NAME
+
+    static SPATIAL_OPERATOR = CswSpatialOperator.DEFAULT_FIELD_NAME
+
+    static BASE_PATH = [SaveCswConfiguration.ID, BaseAction.ARGUMENT]
+
+    static CONFIG_PATH = [BASE_PATH, SOURCE_CONFIG].flatten()
+
+    static ENDPOINT_URL_PATH = [CONFIG_PATH, ENDPOINT_URL].flatten()
+
+    static SERVICE_PID_PATH = [BASE_PATH, PID].flatten()
+
+    static SOURCE_NAME_PATH = [CONFIG_PATH, SOURCE_NAME].flatten()
+
+    static CSW_PROFILE_PATH = [CONFIG_PATH, CSW_PROFILE].flatten()
+
+    static SPATIAL_OPERATOR_PATH = [CONFIG_PATH, SPATIAL_OPERATOR].flatten()
+
+    static EVENT_SERVICE_ADDR_PATH = [CONFIG_PATH, CswSourceConfigurationField.EVENT_SERVICE_ADDRESS_FIELD_NAME].flatten()
+
+    Action saveCswConfiguration
+
+    ConfiguratorFactory configuratorFactory
+
+    Configurator configurator
+
+    ConfigReader configReader
+
+    FederatedSource federatedSource
+
+    def actionArgs
+
+    def federatedSources = []
+
+    def setup() {
+        actionArgs = createCswSaveArgs()
+        configurator = Mock(Configurator)
+        configReader = Mock(ConfigReader)
+        federatedSource = Mock(FederatedSource)
+        configuratorFactory = Mock(ConfiguratorFactory)
+        federatedSource.getId() >> TEST_SOURCENAME
+        federatedSources.add(federatedSource)
+        configuratorFactory.getConfigurator() >> configurator
+        configuratorFactory.getConfigReader() >> configReader
+        saveCswConfiguration = new SaveCswConfiguration(configuratorFactory)
+    }
+
+    def 'test new configuration save successful'() {
```

Comment:
Nit, adding test to the front of all these is kind of repetitive and making it less readable. I think saying something like 'Successfully save new configuration' is a lot more easier on the eyes.